include/secp256k1_schnorrsig_halfagg.h

```diff
@@ -0,0 +1,107 @@
+#ifndef SECP256K1_SCHNORRSIG_HALFAGG_H
+#define SECP256K1_SCHNORRSIG_HALFAGG_H
+
+#include "secp256k1.h"
+#include "secp256k1_extrakeys.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/** Incrementally (Half-)Aggregate a sequence of Schnorr
+ *  signatures to an existing half-aggregate signature.
+ *
+ *  Returns 1 on success, 0 on failure.
+ *  Args:           ctx: a secp256k1 context object.
+ *  In/Out:      aggsig: pointer to the serialized aggregate signature
+ *                       that is input. Will be overwritten by the new
+ *                       serialized aggregate signature.
+ *          aggsig_size: size of the memory allocated in aggsig.
+ *                       Should be large enough to hold the new
+ *                       serialized aggregate signature.
+ *                       I.e., should be a multiple of 32 and should
+ *                       satisfy aggsig_size >= 32*(n_before+n_new+1)
+ *  In:     all_pubkeys: Array of x-only public keys, including both
```

Comment:
nit: This way of writing things is a bit more concise (and no need to define n, which is then unused) but this is a probably a matter of taste. Feel free to pick up or disregard.
```suggestion
 *  In:     all_pubkeys: Array of (n_before + n_new) x-only public keys, including both
```

If you like this, then I suggest doing it also for the other parameters here and for the other functions.